nb

```diff
@@ -6073,7 +6119,15 @@ HEREDOC
 
           if _file_is_encrypted "${_notebook_path}/${__basename}"
           then
-            _indicators+="🔒 "
+            local _file_command_response
+            _file_command_response="$(file "${_notebook_path}/${__basename}")"
```

Comment:
We should test the performance impact of `file`. I've found that `file` can be slow, so in the `_file_is_` functions I try to avoid calling it. [_file_is_encrypted()](https://github.com/xwmx/nb/blob/master/nb#L1784) uses a series of file extension checks and then finally calls `file` with a bunch of options disabled, which I found can boost performance.

Reply:
FWIW, I do continuous benchmarking with the `time` command and am always looking for ways to shave time off operations.

Reply:
Ah ok thanks i'll take a look at it and do the same.

Reply:
Apologies, I neglected to consider specifying a different filename.

Reply:
These are the recent changes I've made:
[https://github.com/ohyhyb/nb/commit/8bdeed34bdd0b1d302d33d763238d29ab20fa5d3](url)

With my additions and benchmarking with 'hyperfine -r 10' these are the average times with 10 items in a notebook:
```
Benchmark #1: ohyhyb/nb
  Time (mean ± σ):     644.1 ms ±   9.5 ms    [User: 319.5 ms, System: 307.6 ms]
  Range (min … max):   627.0 ms … 660.1 ms    10 runs

Benchmark #2: xwmx/nb
  Time (mean ± σ):     618.3 ms ±   6.1 ms    [User: 307.9 ms, System: 296.4 ms]
  Range (min … max):   610.1 ms … 629.1 ms    10 runs
```
I'm by no means an experienced coder so there are likely better ways for doing this, I feel like I'm butchering your code tbh ;)